sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,692 @@
+            if (returnValueWireType == Base64Url.class) {
+                // Mono<Base64Url>
+                responseBodyBytes = new Base64Url(responseBodyBytes).decodedBytes();
+            }
+            result = responseBodyBytes;
+        }  else if (TypeUtil.isTypeOrSubTypeOf(entityType, BinaryData.class)) {
+            // Mono<BinaryData>
+            // The raw response is directly used to create an instance of BinaryData which then provides
+            // different methods to read the response. The reading of the response is delayed until BinaryData
+            // is read and depending on which format the content is converted into, the response is not necessarily
+            // fully copied into memory resulting in lesser overall memory usage.
+            result = response.getSourceResponse().getContent();
+        } else {
+            // Mono<Object> or Mono<Page<T>>
+            result = response.getDecodedBody((byte[]) null);
+        }
+        return result;
+    }
+
+    /**
+     * Handle the provided asynchronous HTTP response and return the deserialized value.
+     *
+     * @param httpDecodedResponse the asynchronous HTTP response to the original HTTP request
+     * @param methodParser the SwaggerMethodParser that the request originates from
+     * @param returnType the type of value that will be returned
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return the deserialized result
+     */
+    private Object handleRestReturnType(final HttpDecodedResponse httpDecodedResponse,
+        final SwaggerMethodParser methodParser,
+        final Type returnType,
+        final Context context,
+        final RequestOptions options) {
+        final HttpDecodedResponse expectedResponse =
+            ensureExpectedStatus(httpDecodedResponse, methodParser, options);
+        //TODO: Wire Tracing and Context Flow synchronously.
+//                .doOnEach(SyncRestProxy::endTracingSpan)
+//                .contextWrite(reactor.util.context.Context.of("TRACING_CONTEXT", context));
+
+        final Object result;
+
+        if (TypeUtil.isTypeOrSubTypeOf(returnType, void.class) || TypeUtil.isTypeOrSubTypeOf(returnType,
+            Void.class)) {
+            // ProxyMethod ReturnType: Void
+//            asyncExpectedResponse.block();
+            result = expectedResponse;
+        } else {
+            // ProxyMethod ReturnType: T where T != async (Mono, Flux) or sync Void
+            // Block the deserialization until a value T is received
+            result = handleRestResponseReturnType(httpDecodedResponse, methodParser, returnType);
+        }
+        return result;
+    }
+
+    // This handles each onX for the response mono.
+    // The signal indicates the status and contains the metadata we need to end the tracing span.
+    private static void endTracingSpan(Signal<HttpDecodedResponse> signal) {
```

Comment:
take a look at https://github.com/kasobol-msft/azure-sdk-for-java/blob/1dfbd178a69c88b2bb7a4bc7367babb242dfa2ef/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java#L693 and it's usage in the original prototype. This should give you idea how to wire an equivalent here.